### ISSUE TITLE
Summary Review Popup Appears After Deleting Recorded Audio, Leading to Null Pointer Exceptions

### ISSUE THREAD

**Issue by hopemadziakapita** [AppHatchery/Fabla-Front-end#270]
## Issue

When a user enters a response either text or audio and saves it, a popup is displayed allowing them to review their response. However, if the response is deleted, the popup still appears, enabling the user to proceed to the summary screen. This leads to null pointer exceptions because the response no longer exists.

-----
### Steps to Reproduce:
1. Navigate to the questions page with `ResponseType.recording` prompt
2. Record/type response
3. Save the response.
4. Delete the response.
5. Observe that the popup still appears, allowing the user to proceed to the review screen.
6. Attempt to proceed to the review summary, leading to a null pointer exception.

---
### Expected Behaviour:
The popup should not appear if the response has been deleted. The application should handle the deletion appropriately and prevent the user from proceeding to the review screen for a non-existent response.

### Actual Behaviour:
The popup appears even after the audio has been deleted, and the user can attempt to review it, causing a null pointer exception.

### Screenshots 
![Screenshot_2024-05-17-11-26-43-808_edu emory audio_diaries_flutter](https://github.com/AppHatchery/Audio-Diaries-Flutter/assets/48337899/cfc4ec18-7d79-4957-b25b-77fe7c5ddd8d)

![Screenshot_2024-05-17-11-19-57-836_edu emory audio_diaries_flutter](https://github.com/AppHatchery/Audio-Diaries-Flutter/assets/48337899/918ff54a-bb13-427b-9bc9-de361faa85a5)

### Possible Solution:
Implement a check to verify the existence of a response before displaying the review popup. If the response is deleted, the popup should not be shown.

**Comment by yago1994** [comment]
Added some extra content to this issue in #271 

Also I think I caught your deleting bug

Steps to reproduce:
- Go through a diary and create a text response
- Delete this text response multiple times (2x - 3x)
- Finish and submit the entry
- Start a new diary
- You should not be able to see the Audio/Text question page and the diary will only have 3/3 Questions available

This is a major issue, I'd like for @Naomi-Nyama and @Wiza-Munthali to jump in on Monday and help diagnose what is happening. It may be exposing an architectural flaw on the way the diaries are stored and displayed for the repeated entries and it's good we caught this early.

**Comment by hopemadziakapita** [comment]
> Added some extra content to this issue in #271
> 
> Also I think I caught your deleting bug
> 
> Steps to reproduce:
> 
> * Go through a diary and create a text response
> * Delete this text response multiple times (2x - 3x)
> * Finish and submit the entry
> * Start a new diary
> * You should not be able to see the Audio/Text question page and the diary will only have 3/3 Questions available
> 
> This is a major issue, I'd like for @Naomi-Nyama and @Wiza-Munthali to jump in on Monday and help diagnose what is happening. It may be exposing an architectural flaw on the way the diaries are stored and displayed for the repeated entries and it's good we caught this early.

This issue was resolved in #264, Did you encounter this after #264 was merged into main?
